snuba/admin/auth_scopes.py

```diff
@@ -0,0 +1,43 @@
+from dataclasses import dataclass
+from enum import Enum
+from typing import Set
+
+
+class AdminRole(Enum):
+    ADMIN = "admin"
+    MEMBER = "member"
+    MEMBER_READ = "member_read"
+
+
+ADMIN_CATEGORY_RESOURCES = {
+    # clickhouse migrations
+    "migrations.all",
+    "migrations.system",
+    "migrations.generic_metrics",
+    "migrations.profiles",
+    "migrations.functions",
+    "migrations.replays",
+    "migrations.querylog",
+    "migrations.test_migration",
+}
+
+
+@dataclass(frozen=True)
+class AuthScope:
+    category: str
+    resource: str
+    role: AdminRole
+
+    def to_str(self) -> str:
+        return f"{self.category}.{self.resource}.{self.role.value}"
```

Comment:
I am a bit confused about this model. 
As I mentioned on the notion doc, most role based access system have at least three entities: 
- resources
- actions
- principals / roles 

There generally is a relationship of ownership between a role, and action and a resource so that a role is granted the permission to execute the action on a resource.
Then you assign roles to a person.

Here I think the resource is composed by the pair `category + resource`. Is this correct ?
I cannot see the action in this model. 
Also does the scope represent the relationship between the three ? 

Since these concepts in role based access control are generally familiar to developers I would suggest we stick to them without introducing new ones. That makes it much easier for somebody who does not know the snuba admin to pick them up without having to understand the custom model.